### PR TITLE
Feature/deployment instances

### PR DESCRIPTION
**Pull request by filso**
New feature - deployment instances tab
image

### CODE REVIEW

File: server/api/controllers/instances/instancesController.js [hunk 1]
```diff
@@ -12,26 +12,33 @@ let Instance = require('models/Instance');
...
 
     if (cluster !== undefined) {
       filter['tag:OwningCluster'] = cluster;
     }
-    if (environment !== undefined) {
-      filter['tag:Environment'] = environment;
+    if (environmentName !== undefined) {
+      filter['tag:Environment'] = environmentName;
+      if (accountName === undefined) {
+        accountName = yield Environment.getAccountNameForEnvironment(environmentName);
```
Comment (Merlin-Taylor) [inline server/api/controllers/instances/instancesController.js]: This is not currently cached.

File: server/api/swagger.yaml [hunk 2]
```diff
@@ -579,6 +580,14 @@ paths:
           in: query
           type: string
           required: false
+        - name: since
```
Comment (Merlin-Taylor) [inline server/api/swagger.yaml]: Could add a format property to _since_ parameter like on line 270?

File: server/commands/services/ToggleTargetStatus.js [hunk 3]
```diff
@@ -18,9 +17,9 @@ function* ToggleTargetStatus(command) {
   let key = `environments/${environment}/roles/${serverRole}/services/${serviceName}/${slice}`;
   let state = yield serviceTargets.getTargetState(environment, { key });
   let service = state.value;
-  let previousStatus = service.hasOwnProperty(SERVICE_ACTION) ? service[SERVICE_ACTION] : SERVICE_INSTALL;
+  let previousStatus = service['Action'] || SERVICE_INSTALL;
```
Comment (Merlin-Taylor) [inline server/commands/services/ToggleTargetStatus.js]: Nice work removing use of `Object.prototoype.hasOwnProperty`

File: server/modules/environment-state/getInstanceState.js [hunk 4]
```diff
@@ -90,7 +90,36 @@ module.exports = function getInstanceState(accountName, environmentName, nodeNam
...
+          DiffWithTargetState: 'Missing',
+          Issues: { Warnings: [], Errors: [] }
+        };
+        missingService.Issues.Warnings.push(`Service that is in target state is missing`);
+        services.push(missingService);
+      }
+    });
+
+    // Secondo, find any services that are present on instance, but not in target state
+    _.each(services, (instanceService) => {
```
Comment (Merlin-Taylor) [inline server/modules/environment-state/getInstanceState.js]: Do we need to handle services that are present in the KV store but marked not to be installed?

File: server/api/controllers/instances/instancesController.js [hunk 5]
```diff
@@ -42,13 +49,54 @@ function getInstances(req, res, next) {
...
+        
+        instance.appendTagsToObject();
+
+        let instanceName = instance.getTag('Name', null);
+        if (instanceName === null || instanceEnvironment === null) {
+          // This instance won't be returned
+          return false;
+        }
+
+        return Environment.getAccountNameForEnvironment(instanceEnvironment).then((accountName) => {
```
Comment (Merlin-Taylor) [inline server/api/controllers/instances/instancesController.js]: Environment.getAccountNameForEnvironment is not currently cached and here called in tight loop over all instances.

File: server/api/swagger.yaml (hunk 2 above)
Comment (filso) [inline server/api/swagger.yaml]: will do, good catch!

**Pull request merged**